external/storm-hbase/src/main/java/org/apache/storm/hbase/trident/windowing/HBaseWindowsStore.java

```diff
@@ -0,0 +1,275 @@
+import java.io.UnsupportedEncodingException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+/**
+ * This class stores entries into hbase instance of the given configuration.
+ *
+ */
+public class HBaseWindowsStore implements WindowsStore {
+    private static final Logger log = LoggerFactory.getLogger(HBaseWindowsStore.class);
+    public static final String UTF_8 = "utf-8";
+
+    private final ThreadLocal<HTable> threadLocalHtable;
+    private Queue<HTable> htables = new ConcurrentLinkedQueue<>();
+    private final byte[] family;
+    private final byte[] qualifier;
+
+    public HBaseWindowsStore(final Configuration config, final String tableName, byte[] family, byte[] qualifier) {
+        this.family = family;
+        this.qualifier = qualifier;
+
+        threadLocalHtable = new ThreadLocal<HTable>() {
+            @Override
+            protected HTable initialValue() {
+                try {
+                    HTable hTable = new HTable(config, tableName);
+                    htables.add(hTable);
+                    return hTable;
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        };
+
+    }
+
+    private HTable htable() {
+        return threadLocalHtable.get();
+    }
+
+    private byte[] effectiveKey(String key) {
+        try {
+            return key.getBytes(UTF_8);
+        } catch (UnsupportedEncodingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public Object get(String key) {
+        WindowsStore.Entry.nonNullCheckForKey(key);
+
+        byte[] effectiveKey = effectiveKey(key);
+        Get get = new Get(effectiveKey);
+        Result result = null;
+        try {
+            result = htable().get(get);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        if(result.isEmpty()) {
+            return null;
+        }
+
+        Kryo kryo = new Kryo();
+        Input input = new Input(result.getValue(family, qualifier));
+        Object resultObject = kryo.readClassAndObject(input);
+        return resultObject;
+
+    }
+
+    @Override
+    public Iterable<Object> get(List<String> keys) {
+        List<Get> gets = new ArrayList<>();
+        for (String key : keys) {
+            WindowsStore.Entry.nonNullCheckForKey(key);
+
+            byte[] effectiveKey = effectiveKey(key);
+            gets.add(new Get(effectiveKey));
+        }
+
+        Result[] results = null;
+        try {
+            results = htable().get(gets);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        Kryo kryo = new Kryo();
+        List<Object> values = new ArrayList<>();
+        for (int i=0; i<results.length; i++) {
```

Comment:
results is an array of Result[] and it can not be reduced to for(int i: results). I did not use for(Result result: results) as I need index to get key as mentioned in the below code. 
@harshach pl let me know if I am missing anything here.

``` java
if(result.isEmpty()) {
                log.error("Got empty result for key [{}]", keys.get(i));
                throw new RuntimeException("Received empty result for key: "+keys.get(i));
            }
```